lib/Arduino_Core_Buddy/cores/arduino/USBSerial.cpp

```diff
@@ -47,7 +47,7 @@ void USBSerial::flush(void) {
     if (enabled)
         tud_cdc_write_flush();
 
-    if (lineBufferUsed) {
+    if (lineBufferUsed && lineBufferHook) {
```

Comment:
generally I prefer to 'escape early' (guarding) rather than 'do if valid' (current state), ie in this case that would mean:
```
if (!lineBufferHook || !lineBufferUsed){
   return;
}

lineBufferHook(&lineBuffer[0], lineBufferUsed);
lineBufferUsed = 0;
```

However It's up to you, both options are valid.